tbsoc/server/data_manager.py

```
@@ -188,3 +189,95 @@ def calculate_tb_bands(self, lambdas_list):
+    def save_hr_file(self, lambdas_list):
+        if self.data_dict is None:
+            raise ValueError("Data not loaded")
+            
+        full_lambdas = np.array(lambdas_list)
+        
+        # 1. Calculate SOC onsite matrix
+        # indices are precalculated in _precalculate or we re-derive them
+        fit_indices = np.where(full_lambdas != 0)[0]
+        
+        if len(fit_indices) == 0:
+            h_soc = 0.0 # No soc
```

Ruff (0.14.10)

Local variable `h_soc` is assigned to but never used

Remove assignment to unused variable `h_soc`

(F841)